server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequest.java

```diff
@@ -84,174 +85,68 @@ public NodesInfoRequest clear() {
+            throw new IllegalStateException("Used an illegal metric: " + metric);
+        }
+        requestedMetrics.add(metric);
         return this;
     }
 
     /**
-     * Should information about plugins be returned
-     * @param plugins true if you want info
-     * @return The request
+     * Add collection of metrics
      */
-    public NodesInfoRequest plugins(boolean plugins) {
-        addOrRemoveMetric(plugins, Metrics.PLUGINS.metricName());
+    public NodesInfoRequest addMetrics(Collection<String> metrics) {
+        if (Metric.allMetrics().containsAll(metrics) == false) {
+            throw new IllegalStateException("Used an illegal metric: " + metrics);
+        }
+        requestedMetrics.addAll(metrics);
         return this;
     }
 
     /**
-     * @return true if information about plugins is requested
+     * Add String array of metrics
      */
-    public boolean plugins() {
-        return Metrics.PLUGINS.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should information about ingest be returned
-     * @param ingest true if you want info
-     */
-    public NodesInfoRequest ingest(boolean ingest) {
-        addOrRemoveMetric(ingest, Metrics.INGEST.metricName());
+    public NodesInfoRequest addMetrics(String... metrics) {
+        addMetrics(Arrays.stream(metrics).collect(Collectors.toSet()));
         return this;
     }
 
     /**
-     * @return true if information about ingest is requested
+     * Remove metric
      */
-    public boolean ingest() {
-        return Metrics.INGEST.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should information about indices (currently just indexing buffers) be returned
-     * @param indices true if you want info
-     */
-    public NodesInfoRequest indices(boolean indices) {
-        addOrRemoveMetric(indices, Metrics.INDICES.metricName());
+    public NodesInfoRequest removeMetric(String metric) {
+        if (Metric.allMetrics().contains(metric) == false) {
+            throw new IllegalStateException("Used an illegal metric: " + metric);
+        }
+        requestedMetrics.remove(metric);
         return this;
     }
 
     /**
-     * @return true if information about indices (currently just indexing buffers)
-     */
-    public boolean indices() {
-        return Metrics.INDICES.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Helper method for adding and removing metrics.
-     * @param includeMetric Whether or not to include a metric.
+     * Helper method for adding and removing metrics. Used when deserializing
+     * a NodesInfoRequest from an ordered list of booleans.
+     *
+     * @param addMetric Whether or not to include a metric.
      * @param metricName Name of the metric to include or remove.
      */
-    private void addOrRemoveMetric(boolean includeMetric, String metricName) {
-        if (includeMetric) {
+    private void optionallyAddMetric(boolean addMetric, String metricName) {
```

Comment:
This method is now only used for backwards-compatible deserialization, and doesn't need to be able to remove metrics anymore.